OpenRA.Mods.Common/Traits/BotModules/BotProductionConsideration.cs

```diff
@@ -0,0 +1,93 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	[Desc("Controls AI production.")]
+	public class BotProductionConsiderationInfo : ConditionalTraitInfo
```

Comment:
`BotProductionComposition` is probably a better name for this.

Reply:
Depends on how you understand the feature. I would welcome it if it could support both large-scale compositions and fine-tuned considerations.

Reply:
Depends on how you understand the feature. I would welcome it if it could support both large-scale compositions and fine-tuned considerations.